Destiny Bond fails on repeated use in Gen 7+

Pawkkie opened the pull request:
## Description
Destiny Bond fails if used multiple turns in a row from Gen 7 onwards

https://bulbapedia.bulbagarden.net/wiki/Destiny_Bond_(move)

## Issue(s) that this PR fixes
#5651 

## **People who collaborated with me in this PR**
freezai for the idea!

## **Discord contact info**
@Pawkkie

AlexOn1ine commented on src/battle_script_commands.c:
```diff
@@ -13144,12 +13144,22 @@ static void Cmd_trychoosesleeptalkmove(void)
+    }
+    else
+    {
+        if (B_UPDATED_MOVE_DATA >= GEN_7)
```
We usually do standalone configs for move effects.

Pawkkie replied:
Oh sure! I don't often do these, fixed :)

hedara90 commented:
Shouldn't Destiny Bond be able to be used if something is used in-between Destiny Bond usages?
Battler goes: Destiny Bond (Success) -> Tackle -> Destiny Bond (Success)
But right not it goes: Destiny Bond (Success) -> Tackle -> Destiny Bond (Fail)

Pawkkie commented:
> Shouldn't Destiny Bond be able to be used if something is used in-between Destiny Bond usages? Battler goes: Destiny Bond (Success) -> Tackle -> Destiny Bond (Success) But right not it goes: Destiny Bond (Success) -> Tackle -> Destiny Bond (Fail)
> 
> E: I don't know how it's supposed to interact if the battler gets flinched, has truant or is otherwise prevented from moving.

LOL this is what I get for jamming in a feature between meetings.

I'll address once I'm actually done for the day and on the other side of dinner :)

AlexOn1ine commented:
> > Shouldn't Destiny Bond be able to be used if something is used in-between Destiny Bond usages? Battler goes: Destiny Bond (Success) -> Tackle -> Destiny Bond (Success) But right not it goes: Destiny Bond (Success) -> Tackle -> Destiny Bond (Fail)
> > E: I don't know how it's supposed to interact if the battler gets flinched, has truant or is otherwise prevented from moving.
> 
> LOL this is what I get for jamming in a feature between meetings.
> 
> I'll address once I'm actually done for the day and on the other side of dinner :)

In this case maybe use LastLandedMoves? You also don't need the new field in this case.

Pawkkie commented:
> In this case maybe use LastLandedMoves? You also don't need the new field in this case.

This was a sick idea that didn't work, but `gLastResultingMoves` did what we wanted `gLastLandedMoves` to do and cleaned it up very nicely :)

The difference between those two fields is not entirely clear to me lol

Fixed!

Pawkkie commented:
1.10 conflicts solved, ready for re-review :)

hedara90 commented:
I have a question about this.
Should the other Gen 7+ behaviors also be implemented in this PR or is it out of scope?
In Gen 8 it's blocked with a custom message if any Dynamax mon is on the field.
And I'm checking some Gen 7 interactions right now

Pawkkie commented:
I'm personally not interested in sorting out the gimmick handling, though if there are other niche cases outside of gimmicks I'd be open to adding them :)

Also fixed conflicts!

AlexOn1ine commented on src/battle_ai_main.c:
```diff
@@ -1875,6 +1875,8 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
                 ADJUST_SCORE(-10);
             break;
         case EFFECT_DESTINY_BOND:
+            if (DoesDestinyBondFail(battlerAtk))
+                ADJUST_SCORE(-10);
             if (gBattleMons[battlerDef].status2 & STATUS2_DESTINY_BOND)
                 ADJUST_SCORE(-10);
```
Does it make sense to also put the status bit check in the function itself? It is kinda what the function is doing according by the name

AlexOn1ine replied:
Sorry, I completely missed the point that the function is also used by the battle engine.
